[CICD] Fuzz test workflow fails

The fuzz test workflow fails with an error:
>   error[E0658]: use of unstable library feature 'proc_macro_is_available'
    --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.33/src/detection.rs:28:21
     |
  28 |     let available = proc_macro::is_available();
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: see issue #71436 <https://github.com/rust-lang/rust/issues/71436> for more information
     = help: add `#![feature(proc_macro_is_available)]` to the crate attributes to enable

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -38,7 +38,7 @@ jobs:
         with:
           override: true
           profile: minimal
-          toolchain: nightly-2021-09-28
+          toolchain: nightly-2021-12-08
 
       - name: Cache dependencies
         uses: Swatinem/rust-cache@v1

--- a/.github/workflows/migration.yml
+++ b/.github/workflows/migration.yml
@@ -19,6 +19,6 @@ jobs:
         with:
           override: true
           profile: minimal
-          toolchain: nightly-2021-09-28
+          toolchain: nightly-2021-12-08
 
       - run: ./scripts/runtime-upgrade/test_runtime_upgrade.sh ${{ github.event.inputs.block }}

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -31,7 +31,7 @@ jobs:
           components: clippy, rustfmt
           override: true
           profile: minimal
-          toolchain: nightly-2021-09-28
+          toolchain: nightly-2021-12-08
 
       - name: Cache Dependencies
         uses: Swatinem/rust-cache@v1
@@ -51,7 +51,7 @@ jobs:
         with:
           override: true
           profile: minimal
-          toolchain: nightly-2021-09-28
+          toolchain: nightly-2021-12-08
 
       - uses: actions-rs/install@v0.1
         with:

--- a/Dockerfile
+++ b/Dockerfile
@@ -18,8 +18,8 @@ RUN apt-get update && \
 
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
     export PATH="$PATH:$HOME/.cargo/bin" && \
-    rustup toolchain install nightly-2021-09-28 && \
-    rustup target add wasm32-unknown-unknown --toolchain nightly-2021-09-28 && \
+    rustup toolchain install nightly-2021-12-08 && \
+    rustup target add wasm32-unknown-unknown --toolchain nightly-2021-12-08 && \
     rustup default stable && \
     cargo build "--$PROFILE" --features "$FEATURES"
 

--- a/runtime/src/lib.rs
+++ b/runtime/src/lib.rs
@@ -66,7 +66,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     impl_name: create_runtime_str!("zeitgeist"),
     authoring_version: 1,
     spec_version: 26,
-    impl_version: 1,
+    impl_version: 2,
     apis: RUNTIME_API_VERSIONS,
     transaction_version: 7,
 };

--- a/rust-toolchain
+++ b/rust-toolchain
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2021-09-28"
+channel = "nightly-2021-12-08"
 components = ["clippy", "rustfmt"]
 profile = "minimal"
 targets = ["wasm32-unknown-unknown"]

--- a/scripts/init.sh
+++ b/scripts/init.sh
@@ -5,8 +5,8 @@ set -e
 echo "*** Initializing WASM build environment"
 
 if [ -z $CI_PROJECT_NAME ] ; then
-   rustup update nightly-2021-09-28
+   rustup update nightly-2021-12-08
    rustup update stable
 fi
 
-rustup target add wasm32-unknown-unknown --toolchain nightly-2021-09-28
+rustup target add wasm32-unknown-unknown --toolchain nightly-2021-12-08

--- a/zrml/court/src/lib.rs
+++ b/zrml/court/src/lib.rs
@@ -408,14 +408,13 @@ mod pallet {
                 }
             }
 
-            let mut best_score;
             let mut iter = scores.iter();
 
-            if let Some(first) = iter.next() {
-                best_score = first;
+            let mut best_score = if let Some(first) = iter.next() {
+                first
             } else {
                 return Err(Error::<T>::NoVotes.into());
-            }
+            };
 
             let mut second_best_score = if let Some(second) = iter.next() {
                 if second.1 > best_score.1 {

--- a/zrml/orderbook-v1/src/benchmarks.rs
+++ b/zrml/orderbook-v1/src/benchmarks.rs
@@ -16,13 +16,7 @@ use zeitgeist_primitives::{constants::BASE, types::Asset};
 
 // Takes a `seed` and returns an account. Use None to generate a whitelisted caller
 fn generate_funded_account<T: Config>(seed: Option<u32>) -> Result<T::AccountId, &'static str> {
-    let acc: T::AccountId;
-
-    if let Some(s) = seed {
-        acc = account("AssetHolder", 0, s);
-    } else {
-        acc = whitelisted_caller();
-    }
+    let acc = if let Some(s) = seed { account("AssetHolder", 0, s) } else { whitelisted_caller() };
 
     let asset = Asset::CategoricalOutcome::<T::MarketId>(0u32.into(), 0);
     let _ = T::Shares::deposit(asset, &acc, BASE.saturating_mul(1_000).saturated_into())?;

--- a/zrml/prediction-markets/src/lib.rs
+++ b/zrml/prediction-markets/src/lib.rs
@@ -488,36 +488,31 @@ mod pallet {
             ensure!(amounts.len() == keep.len(), Error::<T>::NotEnoughAssets);
 
             // Create the correct market
-            let weight_market_creation;
-            match assets.clone() {
-                MarketType::Categorical(category_count) => {
-                    weight_market_creation = Self::create_categorical_market(
-                        origin.clone(),
-                        oracle,
-                        period,
-                        metadata,
-                        creation,
-                        category_count,
-                        mdm,
-                        ScoringRule::CPMM,
-                    )?
-                    .actual_weight
-                    .unwrap_or_else(T::WeightInfo::create_categorical_market)
-                }
-                MarketType::Scalar(range) => {
-                    weight_market_creation = Self::create_scalar_market(
-                        origin.clone(),
-                        oracle,
-                        period,
-                        metadata,
-                        creation,
-                        range,
-                        mdm,
-                        ScoringRule::CPMM,
-                    )?
-                    .actual_weight
-                    .unwrap_or_else(T::WeightInfo::create_scalar_market)
-                }
+            let weight_market_creation = match assets.clone() {
+                MarketType::Categorical(category_count) => Self::create_categorical_market(
+                    origin.clone(),
+                    oracle,
+                    period,
+                    metadata,
+                    creation,
+                    category_count,
+                    mdm,
+                    ScoringRule::CPMM,
+                )?
+                .actual_weight
+                .unwrap_or_else(T::WeightInfo::create_categorical_market),
+                MarketType::Scalar(range) => Self::create_scalar_market(
+                    origin.clone(),
+                    oracle,
+                    period,
+                    metadata,
+                    creation,
+                    range,
+                    mdm,
+                    ScoringRule::CPMM,
+                )?
+                .actual_weight
+                .unwrap_or_else(T::WeightInfo::create_scalar_market),
             };
 
             // Buy a complete set of assets based on the highest number to be deployed
@@ -1346,8 +1341,8 @@ mod pallet {
             let market = T::MarketCommons::market(market_id)?;
             if market.status == MarketStatus::Reported {
                 let report = market.report.ok_or(Error::<T>::MarketIsNotReported)?;
-                MarketIdsPerReportBlock::<T>::mutate(&report.at, |mut ids| {
-                    remove_item::<MarketIdOf<T>>(&mut ids, market_id);
+                MarketIdsPerReportBlock::<T>::mutate(&report.at, |ids| {
+                    remove_item::<MarketIdOf<T>>(ids, market_id);
                 });
             }
             if market.status == MarketStatus::Disputed {
@@ -1356,8 +1351,8 @@ mod pallet {
                     let at = last_dispute.at;
                     let mut old_disputes_per_block = MarketIdsPerDisputeBlock::<T>::get(&at);
                     remove_item::<MarketIdOf<T>>(&mut old_disputes_per_block, market_id);
-                    MarketIdsPerDisputeBlock::<T>::mutate(&at, |mut ids| {
-                        remove_item::<MarketIdOf<T>>(&mut ids, market_id);
+                    MarketIdsPerDisputeBlock::<T>::mutate(&at, |ids| {
+                        remove_item::<MarketIdOf<T>>(ids, market_id);
                     });
                 }
             }
@@ -1495,20 +1490,18 @@ mod pallet {
         fn ensure_market_start_is_in_time(
             period: &MarketPeriod<T::BlockNumber, MomentOf<T>>,
         ) -> DispatchResult {
-            let interval;
-
-            match period {
+            let interval = match period {
                 MarketPeriod::Block(range) => {
                     let interval_blocks: u128 = range
                         .start
                         .saturating_sub(<frame_system::Pallet<T>>::block_number())
                         .saturated_into();
-                    interval = interval_blocks.saturating_mul(MILLISECS_PER_BLOCK.into());
+                    interval_blocks.saturating_mul(MILLISECS_PER_BLOCK.into())
                 }
                 MarketPeriod::Timestamp(range) => {
-                    interval = range.start.saturating_sub(T::MarketCommons::now()).saturated_into();
+                    range.start.saturating_sub(T::MarketCommons::now()).saturated_into()
                 }
-            }
+            };
 
             ensure!(
                 <MomentOf<T>>::saturated_from(interval) >= T::MinSubsidyPeriod::get(),
@@ -1724,17 +1717,10 @@ mod pallet {
 
             let retain_closure =
                 |subsidy_info: &SubsidyUntil<T::BlockNumber, MomentOf<T>, MarketIdOf<T>>| {
-                    let market_ready;
-
-                    // Determine whether the current market is past it's subsidy phase
-                    match &subsidy_info.period {
-                        MarketPeriod::Block(period) => {
-                            market_ready = period.start <= current_block;
-                        }
-                        MarketPeriod::Timestamp(period) => {
-                            market_ready = period.start <= current_time;
-                        }
-                    }
+                    let market_ready = match &subsidy_info.period {
+                        MarketPeriod::Block(period) => period.start <= current_block,
+                        MarketPeriod::Timestamp(period) => period.start <= current_time,
+                    };
 
                     if market_ready {
                         let pool_id = T::MarketCommons::market_pool(&subsidy_info.market_id);
@@ -1883,8 +1869,8 @@ mod pallet {
         ) -> DispatchResult {
             if let Some(last_dispute) = disputes.last() {
                 let at = last_dispute.at;
-                MarketIdsPerDisputeBlock::<T>::mutate(&at, |mut ids| {
-                    remove_item::<MarketIdOf<T>>(&mut ids, market_id);
+                MarketIdsPerDisputeBlock::<T>::mutate(&at, |ids| {
+                    remove_item::<MarketIdOf<T>>(ids, market_id);
                 });
             }
             Ok(())

--- a/zrml/swaps/src/lib.rs
+++ b/zrml/swaps/src/lib.rs
@@ -1649,9 +1649,7 @@ mod pallet {
 
             let params = SwapExactAmountParams {
                 asset_amounts: || {
-                    let asset_amount_out: BalanceOf<T>;
-
-                    if pool.scoring_rule == ScoringRule::CPMM {
+                    let asset_amount_out = if pool.scoring_rule == ScoringRule::CPMM {
                         let balance_out = T::Shares::free_balance(asset_out, &pool_account_id);
                         let balance_in = T::Shares::free_balance(asset_in, &pool_account_id);
                         ensure!(
@@ -1663,7 +1661,7 @@ mod pallet {
                                 .saturated_into(),
                             Error::<T>::MaxInRatio
                         );
-                        asset_amount_out = crate::math::calc_out_given_in(
+                        crate::math::calc_out_given_in(
                             balance_in.saturated_into(),
                             Self::pool_weight_rslt(&pool, &asset_in)?,
                             balance_out.saturated_into(),
@@ -1671,7 +1669,7 @@ mod pallet {
                             asset_amount_in.saturated_into(),
                             pool.swap_fee.ok_or(Error::<T>::PoolMissingFee)?.saturated_into(),
                         )?
-                        .saturated_into();
+                        .saturated_into()
                     } else {
                         let base_asset = pool.base_asset.ok_or(Error::<T>::BaseAssetNotFound)?;
                         ensure!(asset_out == base_asset, Error::<T>::UnsupportedTrade);
@@ -1698,10 +1696,8 @@ mod pallet {
                             T::RikiddoSigmoidFeeMarketEma::cost(pool_id, &outstanding_before)?;
                         let cost_after =
                             T::RikiddoSigmoidFeeMarketEma::cost(pool_id, &outstanding_after)?;
-                        asset_amount_out = cost_before
-                            .checked_sub(&cost_after)
-                            .ok_or(ArithmeticError::Overflow)?;
-                    }
+                        cost_before.checked_sub(&cost_after).ok_or(ArithmeticError::Overflow)?
+                    };
                     ensure!(asset_amount_out >= min_asset_amount_out, Error::<T>::LimitOut);
 
                     Ok([asset_amount_in, asset_amount_out])
@@ -1739,9 +1735,7 @@ mod pallet {
             let params = SwapExactAmountParams {
                 asset_amounts: || {
                     let balance_out = T::Shares::free_balance(asset_out, &pool_account_id);
-                    let asset_amount_in: BalanceOf<T>;
-
-                    if pool.scoring_rule == ScoringRule::CPMM {
+                    let asset_amount_in = if pool.scoring_rule == ScoringRule::CPMM {
                         ensure!(
                             asset_amount_out
                                 <= bmul(
@@ -1753,7 +1747,7 @@ mod pallet {
                         );
 
                         let balance_in = T::Shares::free_balance(asset_in, &pool_account_id);
-                        asset_amount_in = crate::math::calc_in_given_out(
+                        crate::math::calc_in_given_out(
                             balance_in.saturated_into(),
                             Self::pool_weight_rslt(&pool, &asset_in)?,
                             balance_out.saturated_into(),
@@ -1761,7 +1755,7 @@ mod pallet {
                             asset_amount_out.saturated_into(),
                             pool.swap_fee.ok_or(Error::<T>::PoolMissingFee)?.saturated_into(),
                         )?
-                        .saturated_into();
+                        .saturated_into()
                     } else {
                         let base_asset = pool.base_asset.ok_or(Error::<T>::BaseAssetNotFound)?;
                         ensure!(asset_in == base_asset, Error::<T>::UnsupportedTrade);
@@ -1787,10 +1781,8 @@ mod pallet {
                             T::RikiddoSigmoidFeeMarketEma::cost(pool_id, &outstanding_before)?;
                         let cost_after =
                             T::RikiddoSigmoidFeeMarketEma::cost(pool_id, &outstanding_after)?;
-                        asset_amount_in = cost_after
-                            .checked_sub(&cost_before)
-                            .ok_or(ArithmeticError::Overflow)?;
-                    }
+                        cost_after.checked_sub(&cost_before).ok_or(ArithmeticError::Overflow)?
+                    };
 
                     ensure!(asset_amount_in <= max_amount_asset_in, Error::<T>::LimitIn);
                     Ok([asset_amount_in, asset_amount_out])

